ext/ffi/jit.rs

```diff
@@ -0,0 +1,247 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+use crate::tcc::Context;
+use crate::NativeType;
+use crate::Symbol;
+use deno_core::error::AnyError;
+use deno_core::v8;
+use std::ffi::c_void;
+use std::ffi::CString;
+
+macro_rules! cstr {
+  ($st:expr) => {
+    &CString::new($st).unwrap()
+  };
+}
+
+fn native_to_c(ty: &NativeType) -> &'static str {
+  match ty {
+    NativeType::Void => "void",
+    NativeType::U8 => "unsigned char",
+    NativeType::U16 => "unsigned short",
+    NativeType::U32 => "unsigned int",
+    NativeType::U64 | NativeType::USize => "unsigned long",
+    NativeType::I8 => "char",
+    NativeType::I16 => "short",
+    NativeType::I32 => "int",
+    NativeType::I64 | NativeType::ISize => "long",
+    NativeType::F32 => "float",
+    NativeType::F64 => "double",
+    NativeType::Pointer | NativeType::Function => "void*",
+  }
+}
+
+macro_rules! impl_arg_int32 {
+  ($ctx: ident, $ty: ty) => {
+    paste::paste! {
+      unsafe extern "C" fn [<deno_ffi_ $ty>] (
+        info: *const v8::FunctionCallbackInfo,
+        index: i32,
+      ) -> $ty {
+        let scope = &mut v8::CallbackScope::new(&*info);
+        let info = v8::FunctionCallbackArguments::from_function_callback_info(info);
+        info.get(index).int32_value(scope).unwrap() as $ty
```

Comment:
I wonder what it takes to crash this? The algorithm for `int32_value` and `uint32_value` did not have errors for it, I think, but it starts with `Number(value)` which I think does have an error case.

But, I don't think that's exactly an issue as I think the error case is a supremely odd edge case.